pkg/endpoint/endpoint.go

```diff
@@ -421,6 +420,7 @@ func NewEndpointWithState(ID uint16, state string) *Endpoint {
 		controllers:   controller.NewManager(),
 	}
 	ep.UpdateLogger(nil)
+	ep.SetDefaultOpts(option.Config.Opts)
```

Comment:
please set defaults before updating logger, I'm not sure if the update logger users any options from the endpoint or not.